src/containers/Folder/components/InfoContainer/InfoContainer.tsx

```diff
@@ -0,0 +1,59 @@
+import * as S from './InfoContainer.style';
+import useModal from '@/hooks/useModal';
+import { Folder } from '@/types/Folder.types';
+import { IconDelete, IconEdit, IconShare } from '@/public/svgs';
+
+interface Props {
+  defaultFolder: Folder;
+  selectedFolder: Folder;
+  userId: number;
+}
+
+function InfoContainer({ defaultFolder, selectedFolder, userId }: Props) {
+  const [modal, setModal] = useModal({
+    userId: userId,
+    folderName: selectedFolder?.name,
+    folderId: selectedFolder?.id,
+  });
+
+  const setShareModal = () => {
+    setModal('share');
+  };
+
+  const setEditModal = () => {
+    setModal('edit');
+  };
+
+  const setDeleteFolderModal = () => {
+    setModal('deleteFolder');
+  };
```

Comment:
3개의 함수에서 이름과 하는 일이 반복되고 있습니다.
아래처럼 공통 함수를 만들고 key를 전달하는 방식이 되면 더 간단하고 유지보수에 유리할 것 같네요.

```suggestion
  const setModalKey = (key: ModalsKey) => setModal(key)
```